customize_image.sh

```diff
@@ -125,7 +126,12 @@ cp "$BAKERYDIR/templates/rsyslog.conf" "$IMAGEDIR/etc"
 # force HDMI mode even if no HDMI monitor is detected
 sed -r -i -e 's/#hdmi_force_hotplug=1/hdmi_force_hotplug=1/' \
 	  -e 's/#hdmi_drive=2/hdmi_drive=2/' \
-	  "$IMAGEDIR/boot/config.txt"
+	  "$CONFIGTXT"
+
+# Add dwc2 overlay for CM4(S)
+MF_DWC2="\[cm4s\]"
+grep -q  "$MF_DWC2" "$CONFIGTXT" || sed -i "$ a \\\n$MF_DWC2" "$CONFIGTXT"
```

Comment:
Adding the section at the end of the file will break our factory reset.

Factory reset appends the device specific overlay and mac address(es) to the end too. Having `[cm4s]` as last section will break this, because these settings will only get applied to CM4s based devices.

How about adding the section before `[all]` like this:  `sed -i '/^\[all\]/i [cm4s]\n' /boot/config.txt`?

Reply:
good catch